### PR TITLE
Add Images with React sample

### PR DESCRIPTION
**Pull request by metadaddy**
## What does this PR do?

This PR adds a new sample React app and tutorial showing how to use Appwrite Storage.

## Test Plan

At present, in common with the [Todo with React](https://github.com/appwrite/demo-todo-with-react) sample that this app is based on, there is no test code.

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes!

### CODE REVIEW

File: example-storage/.env.example [hunk 1]
```diff
@@ -0,0 +1,4 @@
+REACT_APP_ENDPOINT=
```
Comment (codercatdev) [inline example-storage/.env.example:1]: Can we update this to be 
```
REACT_APP_ENDPOINT=
REACT_APP_PROJECT=
REACT_APP_BUCKET_ID=
```

File: example-storage/CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
+# Contributor Covenant Code of Conduct
```
Comment (codercatdev) [inline example-storage/CODE_OF_CONDUCT.md]: please delete this, I will add a top level that references our org levels

Example
https://github.com/appwrite/.github/blob/main/CODE_OF_CONDUCT.md

File: example-storage/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
+# Contributing
```
Comment (codercatdev) [inline example-storage/CONTRIBUTING.md]: please delete this, I will add a top level that references our org levels

Example
https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md

File: example-storage/LICENSE [hunk 4]
```diff
@@ -0,0 +1,21 @@
+MIT License
```
Comment (codercatdev) [inline example-storage/LICENSE]: please delete this, I will add a top level that references our org levels

File: example-storage/app.json [hunk 5]
```diff
@@ -0,0 +1,24 @@
+{
+  "env": {
```
Comment (codercatdev) [inline example-storage/app.json:2]: Is this in use?

**Comment by metadaddy** [comment]
Hi @christyjacob4 - Tessa, on the appwrite Discord, asked me to add you as a reviewer (and also said this was awesome 🙂). I can't add you directly, so I'm mentioning you instead 👍🏻

**Comment by codercatdev** [comment]
@metadaddy can you update the requested items?

**Comment by metadaddy** [comment]
Hi @codercatdev - I don't think this has had a review yet.

**Review by codercatdev — CHANGES_REQUESTED** [review]
I forgot to commit the review.

File: example-storage/.env.example (hunk 1 above)
Comment (metadaddy) [reply example-storage/.env.example:1]: Good catch - done!

File: example-storage/CODE_OF_CONDUCT.md (hunk 2 above)
Comment (metadaddy) [reply example-storage/CODE_OF_CONDUCT.md]: Done!

File: example-storage/CONTRIBUTING.md (hunk 3 above)
Comment (metadaddy) [reply example-storage/CONTRIBUTING.md]: Done!

File: example-storage/LICENSE (hunk 4 above)
Comment (metadaddy) [reply example-storage/LICENSE]: Done!

File: example-storage/app.json (hunk 5 above)
Comment (metadaddy) [reply example-storage/app.json:2]: Yes - it's for the 'Deploy to Heroku' button. Heroku uses it to prompt the user for env var values. See https://blog.heroku.com/introducing_the_app_json_application_manifest

BTW - I updated `REACT_APP_COLLECTION_ID` to `REACT_APP_BUCKET_ID` here also.

**Comment by metadaddy** [comment]
Hi @codercatdev - thanks for the review! I made all the changes you suggested, squashed the commit and force-pushed.

File: example-storage/README.md [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+
+2. Create a new project in the Appwrite console.
+
+3. Add a new Web App to the project and enter your website's endpoint (`localhost, <project-name>.vercel.app etc`)
+
+    ![Create Web App](https://user-images.githubusercontent.com/20852629/113019434-3c27c900-919f-11eb-997c-1da5a8303ceb.png)
+    
+4. Click **Storage** in the left navigation menu to create a new bucket.
+  * Give the bucket a suitable name, for example, `user-images`, and click **Create**.
+  * Make a note of the **Bucket ID** on the right of the Settings page.
```
Comment (codercatdev) [inline example-storage/README.md]: Can we update this to "click copy Bucket ID button, next to the bucket name"
image

File: example-storage/README.md [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+  * Scroll down the Settings page, and click **Add Role**.
```
Comment (codercatdev) [inline example-storage/README.md]: Add open the settings tab, then scroll down to `Update Permissions`. Click add a role to get started.
image
image

File: example-storage/README.md [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+3. Add a new Web App to the project and enter your website's endpoint (`localhost, <project-name>.vercel.app etc`)
+
+    ![Create Web App](https://user-images.githubusercontent.com/20852629/113019434-3c27c900-919f-11eb-997c-1da5a8303ceb.png)
+    
+4. Click **Storage** in the left navigation menu to create a new bucket.
+  * Give the bucket a suitable name, for example, `user-images`, and click **Create**.
+  * Make a note of the **Bucket ID** on the right of the Settings page.
+  * Scroll down the Settings page, and click **Add Role**. 
+  * In the **Role** dropdown, select `users`, and enable the **Create** permission.
+  * Enable **File Security**.
```
Comment (codercatdev) [inline example-storage/README.md:45]: image

File: example-storage/README.md [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+  * Click **Update** to update the bucket's settings.
+
+    <img width="642" alt="Bucket Settings" src="https://user-images.githubusercontent.com/723517/195677596-f52fd8d0-c94c-4670-9bfb-46565e9b899f.png">
+
+## 🚀 Deploy the Front End
+
+You have two options to deploy the front-end and we will cover both of them here. In either case, you will need to fill in these environment variables that help your frontend connect to Appwrite.
```
Comment (codercatdev) [inline example-storage/README.md:53]: For local copy the `.env.example` file to `.env`.

**Review by codercatdev — CHANGES_REQUESTED** [review]
Just a couple things and we are good to merge.

**Comment by codercatdev** [comment]
@metadaddy were you able to make the above changes?

File: example-storage/README.md (hunk 7 above)
Comment (metadaddy) [reply example-storage/README.md]: Done.

File: example-storage/README.md (hunk 6 above)
Comment (metadaddy) [reply example-storage/README.md]: Done.

File: example-storage/README.md (hunk 8 above)
Comment (metadaddy) [reply example-storage/README.md:45]: Done.

File: example-storage/README.md (hunk 9 above)
Comment (metadaddy) [reply example-storage/README.md:53]: I say this lower down, in the **Run Locally** section. (This was copied verbatim from https://github.com/appwrite/demo-todo-with-react)

**Comment by metadaddy** [comment]
@codercatdev Apologies - was on conference duty for the past couple of weeks. I've applied all the changes you suggested, apart from the last one, since the instruction to copy `.env.example` is already in the README, in the **Run Locally** section.

**Review by codercatdev — APPROVED** [review]

**Pull request merged**